lms/static/scripts/frontend_apps/components/ContentSelector.tsx

```diff
@@ -61,6 +78,15 @@ export default function ContentSelector({
     },
   } = useConfig(['filePicker']);
 
+  // Map the existing content selection to a dialog type and value. We don't
+  // open the corresponding dialog immediately, but do pre-fill the dialog
+  // in some cases if the user does open it.
+  const [initialType, initialValue] = useMemo(
+    () =>
+      initialContent ? extractContentTypeAndValue(initialContent) : [null, ''],
+    [initialContent]
+  );
```

Comment:
See this comment: https://github.com/hypothesis/lms/pull/5157/files#r1149289500

```suggestion
  const [initialType, initialValue] = useMemo(
    () => extractContentTypeAndValue(initialContent),
    [initialContent]
  );
```